src/test/ui/impl-trait/explicit-generic-args-for-impl.stderr

```diff
@@ -0,0 +1,17 @@
+error[E0107]: this function takes at most 1 generic argument but 2 generic arguments were supplied
+  --> $DIR/explicit-generic-args-for-impl.rs:6:5
+   |
+LL |     foo::<str, String>("".to_string());
+   |     ^^^        ------ help: remove this generic argument
+   |     |
+   |     expected at most 1 generic argument
+   |
+note: function defined here, with at most 1 generic parameter: `T`
```

Comment:
Doesn't need to block this PR, but we should note in the tracking issue that there's an opportunity for better diagnostics to explicitly call out that you can't assign generics to `impl Trait`

Reply:
I actually think from a pure syntax point of view it's natural to have this error, because indeed only 1 generic parameters are defined.

Reply:
Oh, the error as-is is *fine*. But it couldn't hurt to have a "note: you aren't allowed to explicitly specify `impl Trait` args as generics" or something of the sort. This is especially true if people assume they can just replace generics with `impl Trait` and then get confused when they can no longer specify generics.